apps/web/src/components/Feed/Posts/PostNftPreview.tsx

```diff
@@ -0,0 +1,102 @@
+import { useRouter } from 'next/router';
+import { Suspense, useCallback } from 'react';
+import { useFragment } from 'react-relay';
+import { graphql } from 'relay-runtime';
+import styled from 'styled-components';
+
+import breakpoints from '~/components/core/breakpoints';
+import FullPageLoader from '~/components/core/Loader/FullPageLoader';
+import { StyledImageWithLoading } from '~/components/LoadingAsset/ImageWithLoading';
+import NftPreview from '~/components/NftPreview/NftPreview';
+import { useModalActions } from '~/contexts/modal/ModalContext';
+import ShimmerProvider from '~/contexts/shimmer/ShimmerContext';
+import { PostNftPreviewFragment$key } from '~/generated/PostNftPreviewFragment.graphql';
+import { StyledVideo } from '~/scenes/NftDetailPage/NftDetailVideo';
+import { LoadableTokenDetailView } from '~/scenes/TokenDetailPage/TokenDetailView';
+
+type Props = {
+  tokenRef: PostNftPreviewFragment$key;
+  tokenSize: number;
+};
+
+export default function PostNftPreview({ tokenRef, tokenSize }: Props) {
+  const token = useFragment(
+    graphql`
+      fragment PostNftPreviewFragment on Token {
+        dbid
+        owner {
+          username
+        }
+        ...NftPreviewFragment
+      }
+    `,
+    tokenRef
+  );
+
+  const { showModal } = useModalActions();
+  const router = useRouter();
+
+  const handleClick = useCallback(() => {
+    const ownerUsername = token.owner?.username;
+
+    const currentUrl = router.asPath;
+    const newUrl = `/${ownerUsername}/token/${token.dbid}`;
+    // set Token Detail Page url without reloading the page. This allows the modal to open quickly, and the user can copy the url
+    window.history.replaceState({ ...window.history.state, as: newUrl, url: newUrl }, '', newUrl);
+
+    showModal({
+      content: (
+        <StyledTokenPreviewModal>
+          <Suspense fallback={<FullPageLoader />}>
+            <LoadableTokenDetailView tokenId={token.dbid} />
+          </Suspense>
+        </StyledTokenPreviewModal>
+      ),
+      isFullPage: true,
+      onClose: () => {
+        // reset url to previous url without reloading the page
+        window.history.replaceState(
+          { ...window.history.state, as: currentUrl, url: currentUrl },
+          '',
+          currentUrl
+        );
+      },
+    });
```

Comment:
**EDIT: looks like you did in fact update `LinkToNftDetailView.tsx` as needed! i still wanna take a cleanup pass**

this should be chill to handle here, but i've had a long-running TODO to consolidate the following logic:

the precedent for opening pages in a separate modal / route like this is handled in `LinkToNftDetailView.tsx` where we use an anchor + href tag. this is coupled with a hook that handles opening the modal in `useDisplayFullPageNftDetailModal`, which has similar logic to here except it uses `push` instead of `replaceState`

ultimately i think we should be using `push` because `replaceState` will produce the behavior where the user clicking the "back" button in the browser will not only close the modal but go to a previous page (potentially bouncing outside of gallery.so)

i'm down to merge this for now, then i can take a pass to consolidate this "open NFT in modal / independent URL" logic everywhere so it's less convoluted and more easily transferrable